google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -332,23 +334,41 @@ String getServiceNameFromAppEngineWebXml(XmlFile appengineWebXml) {
 
   @Override
   public void generateAppYaml(FlexibleRuntime runtime, Module module, Path outputFolderPath) {
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.APP_ENGINE_FLEX_APP_YAML_CREATE)
```

Comment:
ah yes, thats a way to look at it that I hadn't considered. The metrics calls should be very tightly coupled to user action (as close as possible); not to service calls.